src/main/java/dev/openfeature/sdk/AbstractStructure.java

```diff
@@ -6,10 +6,15 @@
 @SuppressWarnings({ "PMD.BeanMembersShouldSerialize", "checkstyle:MissingJavadocType" })
 abstract class AbstractStructure implements Structure {
 
-    protected final Map<String, Value> attributes;
+    private Map<String, Value> attributes;
+
+    @Override
+    public boolean isEmpty() {
+        return attributes == null || attributes.size() == 0;
+    }
 
     AbstractStructure() {
-        this.attributes = new HashMap<>();
+        // intentionally don't initialize the attributes - do this lazily
```

Comment:
Opinion: non-major performance improvement vs. less code later, less readable and possibly less bug prone in fututre?
Some references, from first search results:
[1](https://stackoverflow.com/questions/19587678/java-collection-property-initialisation-best-practice)
[2](https://stackoverflow.com/questions/38124912/best-practice-for-initializing-an-arraylist-field-in-java)
[3](https://stackoverflow.com/questions/26067929/programming-best-practices-to-initialize-or-not-to)
[4](https://stackoverflow.com/questions/6704550/traditional-practice-to-initialize-the-field-members-of-a-class)

Reply:
Reverted for: https://github.com/open-feature/java-sdk/pull/1156#discussion_r1795762686